src/boot/efi/boot.c

```diff
@@ -1878,6 +1964,85 @@ static void config_title_generate(Config *config) {
         }
 }
 
+typedef struct {
+        size_t start_index;
+        size_t end_index;
+} Indices;
+
+static void config_copy_entries(Config *config, ConfigEntry **entries, size_t start_index, size_t end_index, ConfigEntry *group_entry) {
+        assert(config);
+        assert(entries);
+
+        for (size_t i = start_index; i <= end_index; i++) {
+                config_add_entry(config, entries[i]);
+                config->entries[config->n_entries - 1]->group_entry = group_entry;
+        }
+}
+
+static void config_create_groups(Config *config, Indices *indices, size_t group_count){
+        assert(config);
+
+        ConfigEntry **old_entries = config->entries;
+        ConfigEntry **entries = NULL;
+        size_t old_n_entries = config->n_entries;
+        config->entries = entries;
+        config->n_entries = 0;
+        TAKE_PTR(entries);
+
+        config_copy_entries(config, old_entries, 0, indices[0].start_index - 1, NULL);
+        size_t i = 0;
+        for (; i < group_count; i++) {
+                _cleanup_(config_entry_freep) ConfigEntry *group_entry = NULL;
+                group_entry = xnew(ConfigEntry, 1);
+                *group_entry = (ConfigEntry){ .title = xstrdup16(u"More..."), .group_entry = NULL, .type = LOADER_MORE, .id = xstrdup16(u"More...") };
```

Comment:
please add line breaks.

Reply:
If your editor reads `.editorconfig` it would limit the line length automatically. The file at hand violates it at several places already though.